lxd/devices.go

```diff
@@ -301,7 +301,7 @@ func deviceNetlinkListener() (chan []string, chan []string, chan device.USBEvent
 func fillFixedInstances(fixedInstances map[int64][]instance.Instance, inst instance.Instance, effectiveCpus []int64, targetCPUPool []int64, targetCPUNum int, loadBalancing bool) {
 	if len(targetCPUPool) < targetCPUNum {
 		diffCount := len(targetCPUPool) - targetCPUNum
-		logger.Warnf("%v CPUs have been required for pinning, but %v CPUs won't be allocated", len(targetCPUPool), -diffCount)
+		logger.Warn("CPUs have been required for pinning, but some CPUs won't be allocated", logger.Ctx{"available CPUs": len(targetCPUPool), "required CPUs for pinning": targetCPUNum, "difference": -diffCount})
```

Comment:
I think one of the goals for context logging is machine readability; would you be able to eliminate the spaces from the context keys? Happy to be corrected here if spaces are used elsewhere.

Reply:
agreed